### PR TITLE
Take SLF4J MDC into use

### LINKED ISSUE #569: Include run ID in all log rows caused by the run
Log rows like `2024-07-02 11:12:48.824 [pool-1-thread-4] level=INFO  org.ossreviewtoolkit.evaluator.Rule - 	Rule skipped.` would be much more useful if logging always printed the run ID.

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Take [SLF4J's MDC](https://www.slf4j.org/manual.html#mdc) into use by adding some initial entries in all components which can be extended later on. Also fix several issues with the propagation of the MDC that I could find during local testing.

See the commit messages for details.

This PR fixes #569 and is a preparation for implementing a solution for #168.

### CODE REVIEW

**Review by MarcelBochtler — COMMENTED** [review]
Looks good to me, aside from the two minor nits.
Due to the sheer size of this PR, I won't approve it yet, to give others a chance to review.

File: workers/advisor/build.gradle.kts
Comment (sschuberth) [inline workers/advisor/build.gradle.kts:49]: Commit message:
- Typo in "in*i*tial"

File: workers/config/src/main/kotlin/ConfigComponent.kt
Comment (sschuberth) [inline workers/config/src/main/kotlin/ConfigComponent.kt:49]: Unrelated changes in this file.

File: transport/rabbitmq/src/main/kotlin/RabbitMqMessageReceiverFactory.kt
Comment (sschuberth) [inline transport/rabbitmq/src/main/kotlin/RabbitMqMessageReceiverFactory.kt:30]: Commit message:
- Typo in "w*h*ere the factory was created".

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: utils/logging/src/main/kotlin/LoggingUtils.kt
Comment (mnonnenmacher) [reply utils/logging/src/main/kotlin/LoggingUtils.kt:38]: I don't want to add a dependency `ort-utils` everywhere because it brings in unrelated dependencies like `software.amazon.awssdk:s3` and `org.apache.commons:commons-compress`.

File: utils/logging/src/main/kotlin/LoggingUtils.kt
Comment (mnonnenmacher) [reply utils/logging/src/main/kotlin/LoggingUtils.kt:46]: I'd rather keep the API specific and leave it to the caller to decide how other types should be converted to strings. Using `Any` here would make it easy to accidentally pass an object where the string representation blows up the MDC or which does not have a proper `toString()` implementation.

File: utils/logging/src/main/kotlin/LoggingUtils.kt
Comment (sschuberth) [reply utils/logging/src/main/kotlin/LoggingUtils.kt:38]: And anyway it just occurred to me that ORT's version adds `org.apache.logging.log4j.kotlin.CoroutineThreadContext`, whereas this adds `kotlinx.coroutines.slf4j.MDCContext`, due to the different logging APIs used in ORT and ORT server. We should probably align that at some point, but that's a different topic.

File: gradle/libs.versions.toml
Comment (sschuberth) [reply gradle/libs.versions.toml:104]: IMO it's still a bit arbitrary which dependencies are added here vs. in the next commit, so I whould have simply squashed the first two commits. But I'll leave it to you.

**Review by sschuberth — APPROVED** [review]

File: utils/logging/src/main/kotlin/LoggingUtils.kt
Comment (mnonnenmacher) [reply utils/logging/src/main/kotlin/LoggingUtils.kt:38]: Yes, I think I will change the workers to also to use the ORT Server version for consistency later on. But it still works as explained in the commit message:

```
    Note that the server uses the SLF4J API to access the MDC while ORT uses
    the Log4j API. This still works because with `log4j-to-slf4j` on the
    classpath, Log4j uses SLF4J's MDC implementation.
```

File: gradle/libs.versions.toml
Comment (mnonnenmacher) [reply gradle/libs.versions.toml:104]: I'll merge it as is because I'm eager to start testing this on our infrastructure as so far I could only test with Docker Compose.

**Pull request merged**
